lib/chef/application/client.rb

```diff
@@ -280,7 +287,14 @@ def reconfigure
       FileUtils.mkdir_p(Chef::Config.chef_repo_path)
       tarball_path = File.join(Chef::Config.chef_repo_path, 'recipes.tgz')
       fetch_recipe_tarball(Chef::Config[:recipe_url], tarball_path)
-      result = shell_out!("tar zxvf #{tarball_path} -C #{Chef::Config.chef_repo_path}")
+      if Chef::Config[:github]
+        cookbooks_dir = File.join(Chef::Config.chef_repo_path, 'cookbooks')
+        FileUtils.rm_rf(cookbooks_dir, :secure => true)
```

Comment:
After an errant spec test wiped out my chef github checkout (and could have easily wiped out $HOME or other nastiness) I'm incredibly :-1: on this rm_rf being here.

See also http://www.theregister.co.uk/2015/01/17/scary_code_of_the_week_steam_cleans_linux_pcs/